### ISSUE TITLE
Mask applied twice for metrics with 'weighted' average option

### ISSUE THREAD

**Issue by yazanobeidi** [pytorch/torcheval#141]
### 🐛 Describe the bug

Hi, I wonder if I caught a small bug. When using the "weighted" average option for both multi-class `f1` and `recall`, it appears the mask is applied twice. First [when applying the mask](https://github.com/pytorch/torcheval/blob/main/torcheval/metrics/functional/classification/f1_score.py#L212-L216), then [when doing the weighted average](https://github.com/pytorch/torcheval/blob/main/torcheval/metrics/functional/classification/f1_score.py#L231). I could be wrong. Please let me know if this is the case. I can submit a PR, or feel free to do it. 

This is coming up when the `num_classes` is larger than the number of unique values in both input and target.

Unweighted works fine:
```python3
import torch
from torcheval.metrics.functional import multiclass_f1_score
input = torch.tensor([0, 2, 1, 4])
target = torch.tensor([0, 1, 2, 4])
multiclass_f1_score(input, target, num_classes=5)

tensor(0.5000)
```

This is fine
```python3
input = torch.tensor([0, 2, 1, 4])
target = torch.tensor([0, 1, 2, 3])
multiclass_f1_score(input, target, num_classes=5, average='weighted')

tensor(0.2500)
```

This is not
```python3
input = torch.tensor([0, 2, 1, 4])
target = torch.tensor([0, 1, 2, 4])
multiclass_f1_score(input, target, num_classes=5, average='weighted')

WARNING:root:Warning: Some classes do not exist in the target. F1 scores for these classes will be cast to zeros.

---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
Cell In[15], line 3
      1 input = torch.tensor([0, 2, 1, 4])
      2 target = torch.tensor([0, 1, 2, 4])
----> 3 multiclass_f1_score(input, target, num_classes=5, average='weighted')

File /python3.11/site-packages/torch/utils/_contextlib.py:115, in context_decorator.<locals>.decorate_context(*args, **kwargs)
    112 @functools.wraps(func)
    113 def decorate_context(*args, **kwargs):
    114     with ctx_factory():
--> 115         return func(*args, **kwargs)

File /python3.11/site-packages/torcheval/metrics/functional/classification/f1_score.py:115, in multiclass_f1_score(input, target, num_classes, average)
    111 _f1_score_param_check(num_classes, average)
    112 num_tp, num_label, num_prediction = _f1_score_update(
    113     input, target, num_classes, average
    114 )
--> 115 return _f1_score_compute(num_tp, num_label, num_prediction, average)

File /python3.11/site-packages/torcheval/metrics/functional/classification/f1_score.py:228, in _f1_score_compute(num_tp, num_label, num_prediction, average)
    226     return f1.mean()
    227 elif average == "weighted":
--> 228     return (f1 * (num_label[mask] / num_label.sum())).sum()
    229 else:  # average is None
    230     return f1

IndexError: The shape of the mask [5] at index 0 does not match the shape of the indexed tensor [4] at index 0
```

### Versions

NA

**Comment by bobakfb** [comment]
Hi Yazan @JKSenthil  is taking a look!

**Comment by JKSenthil** [comment]
Hi @yazanobeidi, you are absolutely right, the mask is being incorrectly applied twice there. Feel free to submit a PR and we'll review it!

**Comment by bobakfb** [comment]
Fixed in #145
